### PR TITLE
Add support for customize PropertyKey validation function

### LINKED ISSUE #10966: Add a isFile() option to PropertyKey.Builder
**Is your feature request related to a problem? Please describe.**
Multiple file properties don't do validations that the file exists or is accessible. Flagging properties that specify file paths (alluxio.underfs.hdfs.keytab.file) with the option could enabled launch-time checks for validity.

**Describe the solution you'd like**
Add a new option in PropertyKey.Builder like .isFile()

**Describe alternatives you've considered**
N/A

**Urgency**
P1 for User Experience

**Additional context**
Might also be nice to have extra validations for mount properties at runtime (matching file checksums on all nodes etc.)

### PR DESCRIPTION
**Pull request by rongrong**
### What changes are proposed in this pull request?

Fix #10966

### Why are the changes needed?

Please clarify why the changes are needed. For instance,
  1. If you propose a new API, clarify the use case for a new API.
  2. If you fix a bug, describe the bug.

### Does this PR introduce any user facing changes?

Please list the user-facing changes introduced by your change, including
  1. change in user-facing APIs
  2. addition or removal of property keys
  3. webui

### CODE REVIEW

File: core/common/src/main/java/alluxio/conf/InstancedConfiguration.java [hunk 1]
```diff
@@ -202,11 +202,11 @@ public void set(PropertyKey key, Object value) {
    * @param source the source of the the properties (e.g., system property, default and etc)
    */
   public void set(@Nonnull PropertyKey key, @Nonnull Object value, @Nonnull Source source) {
-    Preconditions.checkArgument(key != null && value != null && !value.equals(""),
-        String.format("The key value pair (%s, %s) cannot be null", key, value));
     Preconditions.checkArgument(!value.equals(""),
         String.format("The key \"%s\" cannot be have an empty string as a value. Use "
             + "ServerConfiguration.unset to remove a key from the configuration.", key));
+    Preconditions.checkState(key.validateValue(value),
+        String.format("Invalid valid for property key %s: %s", key, value));
```
Comment (apc999) [inline core/common/src/main/java/alluxio/conf/InstancedConfiguration.java:209]:
```suggestion
    Preconditions.checkState(key.validateValue(value),
        "Invalid valid for property key %s: %s", key, value);
```

**Review by apc999 — COMMENTED** [review]
Can you add UT to `PropertyKeyTest.java`?
Otherwise LGTM

**Comment by apc999** [comment]
```
Error: 8.183 [ERROR] Failures: 
Error: 8.184 [ERROR] alluxio.master.AlluxioMasterProcessTest.stopAfterStandbyTransition[0]
Error: 8.186 [ERROR]   Run 1: AlluxioMasterProcessTest.stopAfterStandbyTransition:209
Error: 8.187 [ERROR]   Run 2: AlluxioMasterProcessTest.stopAfterStandbyTransition:209
```

**Comment by apc999** [comment]
> s this related? When I run the test suite locally it fails sometimes but not consistent. And the failure is on `assertFalse(master.isRunning())`.

I think this test is simply flaky.

**Review by apc999 — APPROVED** [review]

**Review by maobaolong — COMMENTED** [review]
This feature look nice, would you please fix the style issue?

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**